README.adoc

```diff
@@ -118,6 +121,20 @@ see the step doc for more documentation about he allowed versions
 infra.stashJenkinsWar("2.110")
 ----
 
+=== buildPluginWithGradle()
+
+Builds a Jenkins plugin using Gradle.
+The implementation follows the standard build/test/archive pattern. 
+The method targets compatibility with [Gradle JPI Plugin](https://github.com/jenkinsci/gradle-jpi-plugin),
+and it may not work for other use-cases.
+
+The interface is similar to `buildPlugin()`, but some features are not supported.
```

Comment:
The interface should be documented properly. Otherwise if would be really hard for a user to find out how to use this new step.